includes/Classifai/Providers/Azure/TextToSpeech.php

```diff
@@ -0,0 +1,553 @@
+		$settings             = \Classifai\get_azure_text_to_speech_settings();
+		$supported_post_types = isset( $settings['post_types'] ) ? $settings['post_types'] : array();
+
+		$supported_post_types = array_keys(
+			array_filter(
+				$supported_post_types,
+				function( $post_type ) {
+					return ! is_null( $post_type );
+				}
+			)
+		);
+
+		register_rest_field(
+			$supported_post_types,
+			'classifai_synthesize_speech',
+			array(
+				'get_callback'    => function( $object ) {
+					$process_content = get_post_meta( $object['id'], self::SYNTHESIZE_SPEECH_KEY, true );
+					return ( 'no' === $process_content ) ? 'no' : 'yes';
+				},
+				'update_callback' => function ( $value, $object ) {
+					$value = ( 'no' === $value ) ? 'no' : 'yes';
+					return update_post_meta( $object->ID, self::SYNTHESIZE_SPEECH_KEY, $value );
+				},
+				'schema'          => [
+					'type'    => 'string',
+					'context' => [ 'view', 'edit' ],
+				],
+			)
+		);
+
+		register_rest_field(
+			$supported_post_types,
+			'classifai_post_audio_id',
+			array(
+				'get_callback' => function( $object ) {
+					$post_audio_id = get_post_meta( $object['id'], self::AUDIO_ID_KEY, true );
+					return (int) $post_audio_id;
+				},
+				'schema'       => [
+					'type'    => 'integer',
+					'context' => [ 'view', 'edit' ],
+				],
+			)
+		);
+	}
+
+	/**
+	 * Adds audio controls to the post that has speech sythesis enabled.
+	 *
+	 * @param string $content Post content.
+	 * @return string
+	 */
+	public function render_post_audio_controls( $content ) {
+		global $post;
+
+		if ( ! $post ) {
+			return $content;
+		}
+
+		if ( apply_filters( 'classifai_disable_post_to_audio_block', false, $post->ID ) ) {
+			return $content;
+		}
+
+		wp_enqueue_script(
+			'classifai-post-audio-player-js',
+			CLASSIFAI_PLUGIN_URL . '/dist/post-audio-controls.js',
+			array(),
+			CLASSIFAI_PLUGIN_VERSION,
+			true
+		);
+
+		wp_enqueue_style(
+			'classifai-post-audio-player-css',
+			CLASSIFAI_PLUGIN_URL . '/dist/post-audio-controls.css',
+			array(),
+			CLASSIFAI_PLUGIN_VERSION,
+			'all'
+		);
+
+		$audio_attachment_id  = (int) get_post_meta( $post->ID, self::AUDIO_ID_KEY, true );
+
+		if ( ! $audio_attachment_id ) {
+			return $content;
+		}
+
+		$audio_timestamp      = (int) get_post_meta( $post->ID, self::AUDIO_TIMESTAMP_KEY, true );
+		$audio_attachment_url = sprintf(
+			'%1$s?ver=%2$s',
+			wp_get_attachment_url( $audio_attachment_id ),
+			filter_var( $audio_timestamp, FILTER_SANITIZE_NUMBER_INT )
+		);
+
+		ob_start();
+
+		?>
+			<div>
+				<div class='classifai-listen-to-post-wrapper'>
+					<div class="class-post-audio-controls">
+						<span class="dashicons dashicons-controls-play"></span>
+						<span class="dashicons dashicons-controls-pause"></span>
+					</div>
+					<div class='classifai-post-audio-heading'>
+						<?php
+							$list_to_post_text = apply_filters( 'classifai_listen_to_this_post_text', esc_html__( 'Listen to this post', 'classifai' ) );
```

Comment:
Can we add a docblock for this filter?

Reply:
Would also be great to not hardcode the post type here, as this isn't accurate if someone uses this on a post type other than `post`